File: demo-backend/src/main/java/com/example/demo/admin/service/AdminService.java
package com.example.demo.admin.service;

import java.util.List;

import com.example.demo.admin.dto.AdminDashboardResponse;
import com.example.demo.admin.dto.AdminUserUpdateRequest;
import com.example.demo.user.dto.UserResponse;

public interface AdminService {

    AdminDashboardResponse getDashboard();

    List<UserResponse> getUsers();

    UserResponse updateUser(Long userId, AdminUserUpdateRequest request);
}

File: demo-frontend/src/services/adminService.js
import { httpClient, unwrapData } from './http';

export const adminService = {
  async getDashboard() {
    const response = await httpClient.get('/api/admin/dashboard');
    return unwrapData(response);
  },

  async getUsers() {
    const response = await httpClient.get('/api/admin/users');
    return unwrapData(response);
  },

  async updateUser(userId, payload) {
    const response = await httpClient.put(`/api/admin/users/${userId}`, payload);
    return unwrapData(response);
  },
};
